### ISSUE TITLE
Cannot set SO_PASSCRED for a net.UnixConn

### ISSUE THREAD

**Issue by markdingo** [golang/go#36293]
In issue #1101 it is said that
"Passing credentials is already supported: https://golang.org/src/syscall/creds_test.go?h=TestSCMCredentials#L21

Please close this issue."

While that is true, the problem is that that test is construction a Conn from fds. If you start with a Conn (say via an AcceptUnix()) there is no way to get to the fd to call setsockopt.

In other words SO_PASSCRED cannot be set for UnixConn and thus the credentials can never be received.

In short, a go server using UnixConns can never check the credentials of clients.

Issue #1101 is frozen so I couldn't comment there. Sorry if this is the wrong way of re-activating it.

**Comment by ianlancetaylor** [comment]
You can use the `SyscallConn` method to get a `syscall.RawConn` and then use the `Control` method to set `SO_PASSCRED`.

**Comment by markdingo** [comment]
And so you can. Thanks for that, Ian. There's no way I would have worked that out for myself.
